### DISCUSSION TITLE
Connect to multiple mongo databases dynamically

### DISCUSSION THREAD

**Discussion by shyamchandranmec** [gofr-dev/gofr#64] [Q&A] [answered]
I have a requirement to connect to multiple Mongo databases dynamically. Depending on the request parameters I have to connect to a different database each time. How to achieve this?

**Comment by aryanmehrotra** [comment, marked as answer]
To have multiple connections you have to use the `datastore.GetNewMongoDB` function, it will give you the object with methods to the new database.
Your existing connection will not be replaced by this.

**Comment by ace3** [comment]
how about multiple SQL (mysql,etc) connection.

i don't see the similar method for the SQL.

**Comment by VipulZopSmart** [comment]
@ace3 you can check this: https://github.com/gofr-dev/gofr/blob/development/pkg/datastore/db.go#L121

**Comment by ace3** [comment]
ahh, i see. thank you.

**Comment by ace3** [comment]
i just tried the gofr.
i have a question. what is the best way to declare a new SQL client & to consume it ?

right now i declare the variable of the new SQL client in the main() function.
and consume it from the handler/repository.

**Reply by aryanmehrotra** [reply]
Hey, you can use configs to connect to a database as given in the example using-postgres, you would need to change the dialect.
Please refer to the following link
https://gofr.dev/docs/v1/quick-start/connecting-mysql

**Reply by ace3** [reply]
sorry for my misleading statement.

my question is, what is the best way, to add the 2nd database connection, and to call it.

for the primary database(sql) , we can get it from the context
```
c.DB()
```

is it possible to add the 2nd database as additional context, like `c.DB2()`

**Reply by vipul-rawat** [reply]
We would recommend writing a new micro-service, as we believe that is a proper way to deal with things.
